notifications/notifications.go

```diff
@@ -0,0 +1,88 @@
+package notifications
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+// Sender contains the configuration information to send events to the notification service
+type Sender struct {
+	URL string
+}
+
+// EmailMessage contains the required fields for formatting email messages
+type EmailMessage struct {
+	Subject string `json:"subject"`
+	Body    string `json:"body"`
+}
+
+// SlackMessage contains the required fields for formatting slack messages
+type SlackMessage struct {
+	Text string `json:"text"`
+}
+
+// BrowserMessage contains the required fields for formatting browser notifications
+type BrowserMessage struct {
+	Type      string    `json:"type"`
+	Text      string    `json:"text"`
+	Timestamp time.Time `json:"timestamp"`
+}
+
+// Message contains the mappings for formatting notification messages
+type Message struct {
+	Browser BrowserMessage `json:"browser"`
+	Slack   SlackMessage   `json:"slack"`
+	Email   EmailMessage   `json:"email"`
+}
+
+type event struct {
+	Type       string    `json:"type"`
+	InstanceID string    `json:"instance_id"`
+	Timestamp  time.Time `json:"timestamp"`
+	Messages   *Message  `json:"messages"`
+}
+
+// CreateSender creates a Sender
+func CreateSender(url string) Sender {
+	return Sender{
+		URL: url,
+	}
+}
+
+// SendEvent sends an event to the notification service.
+func (s *Sender) SendEvent(et string, instance string, t time.Time, msg *Message) error {
```

Comment:
This being a public api, it's not clear what `et` is. It should be spelt out in the comment above, or at the very least spelt out in full (`eventType` instead of `et`).